azurelinuxagent/ga/exthandlers.py

```diff
@@ -969,38 +968,53 @@ def report_ext_handlers_status(self, incarnation_changed=False):
 
         self.write_ext_handlers_status_to_info_file(vm_status)
 
-    @staticmethod
-    def write_ext_handlers_status_to_info_file(vm_status):
-        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE)
+    def write_ext_handlers_status_to_info_file(self, vm_status):
+
+        status_path = os.path.join(conf.get_lib_dir(), AGENT_STATUS_FILE.format(self.last_etag))
 
         agent_details = {
             "agent_name": AGENT_NAME,
-            "current_version": str(CURRENT_VERSION),
             "goal_state_version": str(GOAL_STATE_AGENT_VERSION),
-            "distro_details": "{0}:{1}".format(DISTRO_NAME, DISTRO_VERSION),
-            "last_successful_status_upload_time": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime()),
```

Comment:
I also dont see an alternative for this in the final fields, what did you replace it with?

Reply:
I was assuming the "processed_time" we collect under "goal_state_aggregate_status " . But thinking about it , if goal_state_aggregate_status is not available then we will miss it. so added it back at line no 982

Reply:
did you add last_successful_status_upload_time back?

Reply:
It now comes from the status_blob_data as "timestampUTC". The new data follows the same structure of the status_blob minus 'formattedMessage' and 'substatus' from each handlerStatus